compiler/src/iree/compiler/GlobalOptimization/Convert1X1FilterConv2DToMatmul.cpp

```diff
@@ -55,19 +57,11 @@ class Convert1x1FilterConvToMatmul : public OpRewritePattern<Conv2DOpType> {
     const int owIndex = isNHWC ? 2 : 3;
     const int ocIndex = isNHWC ? 3 : 1;
 
-    bool isInputHWDynamic = ShapedType::isDynamic(inputShape[ohIndex]) &&
-                            ShapedType::isDynamic(inputShape[owIndex]);
-
-    // We cannot merge the width and height if they are both dynamic as we
-    // cannot expand them back to their dynamic values.
-    if (isInputHWDynamic)
-      return failure();
-
     if (filterShape[khIndex] != 1 || filterShape[kwIndex] != 1)
       return failure();
 
     // TODO(ataei): Support conversion to linalg.batch_matmul.
-    if (inputShape[0] != 1)
+    if (isNCHW && inputShape[0] != 1)
```

Comment:
See my comment on the linked issue: https://github.com/iree-org/iree/issues/18710#issuecomment-2407661454

I think we should generate a linalg.generic with multiple n contraction dimensions for NCHW.

Reply:
Thank you, I'll take a look